clippy_lints/src/lifetimes.rs

```diff
@@ -59,6 +59,40 @@ declare_clippy_lint! {
      would allow omitting them"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for lifetime annotations which can be replaced with anonymous lifetimes (`'_`).
+    ///
+    /// ### Why is this bad?
+    /// The additional lifetimes can make the code look more complicated.
+    ///
+    /// ### Known problems
+    /// - We bail out if the function has a `where` clause where lifetimes
+    /// are mentioned due to potential false positives.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// use std::str::Chars;
+    ///
+    /// fn f<'a>(x: &'a str) -> Chars<'a> {
+    ///     x.chars()
+    /// }
+    /// ```
+    ///
+    /// Use instead:
+    /// ```no_run
+    /// use std::str::Chars;
+    ///
+    /// fn f(x: &str) -> Chars<'_> {
+    ///     x.chars()
+    /// }
+    /// ```
+    #[clippy::version = "1.84.0"]
+    pub ELIDABLE_LIFETIME_NAMES,
+    pedantic,
```

Comment:
This PR does two things at once: it not only splits the lint, but also moves part of it from `complexity` to `pedantic`. Since people would be able to silence the `elidable_lifetime_names`, shouldn't it stay into `complexity`?

My rationale is that an unused variable will get flagged if it doesn't start with `_`, why shouldn't we continue to do the same with unused lifetimes?

Reply:
Since a good number of people want to silence it we generally take that as an indicator that it should be moved into `pedantic`

I don't think it's as severe as unused variables since the lifetimes are still used even though they could be elided